tomni/cytosmart_data_format/annotations/ellipse/main.py

```diff
@@ -1,42 +1,133 @@
-    def __init__(self, radius: Point, center: Point, rotation: float):
+    def __init__(
+        self,
+        radius: Point,
+        center: Point,
+        rotation: float,
+        id: str,
+        label: str,
+        children: List[Annotation],
+        parents: List[Annotation],
+    ):
+        super().__init__(id, label, children, parents)
         self._radius: Point = radius
         self._center: Point = center
         self._rotation: float = rotation
 
-        self._circularity = None
-        pass
+        self._circularity: float = None
+        self._area: float = None
+        self._perimeter: float = None
+        self._aspect_ratio: float = None
+
+    @property
+    def label(self):
+        return super().label
 
-    @property.getter
+    @label.setter
+    def label(self, value) -> None:
+        super().label = value
+
+    @property
     def radius(self) -> Point:
-        """x and y
-        """
-        pass
+        return self._radius
 
-    @property.getter
+    @property
     def center(self) -> Point:
-        """x and y
-        """
-        pass
+        return self._center
 
-    @property.getter
+    @property
     def rotation(self) -> float:
-        """ rotation in degree.
-        """
-        pass
+        return self._rotation
 
-    @property.getter
-    def get_circluratiy(self) -> str:
+    @property
+    def circularity(self) -> float:
+        """Circularity described by 4 * pi * Area / Perimeter**2.
+
+        Returns:
+            float: Ellipse's circularity.
+        """
         if not self._circularity:
-            # FIXME: Probably some helper function to get contours or simply tomni.
-            self._circularity = circularity([[0]])
-            return self._circularity
-        else:
-            return self._circularity
+            self._calculate_circularity()
+
+        return self._circularity
+
+    @property
+    def area(self) -> float:
+        """Area described by pi * radiusX * radiusY.
+
+        Returns:
+            float: Ellipse's area.
+        """
+        if not self._area:
+            self._calculate_area()
+
+        return self._area
+
+    @property
+    def perimeter(self) -> float:
+        """Perimeter described by 2*pi*sqrt((radiusX**2 + radiusY**2) / 2).
 
+        Returns:
+            float: Ellipse's perimeter.
+        """
+        if not self._perimeter:
+            self._calculate_perimeter()
+
+        return self._perimeter
+
+    @property
+    def aspect_ratio(self) -> float:
+        """Ratio between minor and major axis in this case radiusX * 2 / radiusY * 2.
+
+        Returns:
+            float: Ellipse's aspect ratio.
+        """
+        if not self._aspect_ratio:
+            self._calculate_aspect_ratio()
+
+        return self._aspect_ratio
+
+    def to_dict(self) -> dict:
+
+        dict_ellipse = {
+            "type": "ellipse",
+            "center": asdict(self.center),
+            "radius": asdict(self.radius),
```

Comment:
You are changing the Standard format of the JSON!!!!
Please don't unless you discussed it